Clarify whether core packages are required to make bug-fix releases (#297)
scientific-python/specs: spec-0000/index.md

## Description

> This SPEC recommends that all projects across the Scientific Python ecosystem adopt a common time-based policy for dropping support for core packages depended upon by those projects, as well as older versions of Python.

Reviewer: ```suggestion
This SPEC recommends that all projects across the Scientific Python ecosystem adopt a common time-based policy for dropping dependencies. From the perspective of this SPEC, the dependencies in question are core packages as well as older Python versions.
```

## Description

> This SPEC recommends that all projects across the Scientific Python ecosystem adopt a common time-based policy for dropping support
> for older Python and for support for older versions of core packages depended upon by those projects.

Reviewer: ```suggestion
This SPEC recommends that all projects across the Scientific Python ecosystem adopt a common time-based policy for dropping support for core packages depended upon by those projects, as well as older versions of Python.
```

1. Support for Python versions be dropped **3 years** after their initial release.
2. Support for core package dependencies be dropped **2 years** after their initial release.

{{< admonition note >}}
Core packages may or may not decide to provide bug fix releases during the full 2 year period after release.
Therefore, projects may occasionally want to drop support for core package dependencies earlier than recommended by this SPEC.
For instance, if a newer minimum version of a core package is needed by a project due to a critical bug fix,
which is not backported to older versions.
{{< /admonition >}}

### Core Project Endorsement

<!--
Briefly discuss what it means for a core project to endorse this SPEC.
-->

Core project endorsing this SPEC means that those projects encourage all projects across the Scientific Python ecosystem

> to limit how long they support older Python versions and older core packages versions.

Reviewer 2: This SPEC only has a table for the core projects, but the driving philosophy is true for all dependencies I believe.

```suggestion
to limit how long they support older Python versions and older dependency versions.
```

1. Support for Python versions be dropped **3 years** after their initial release.
2. Support for core package dependencies be dropped **2 years** after their initial release.

{{< admonition note >}}

> Core packages may or may not to provide bug fix releases during the full 2 year period after release.

Reviewer 3: ```suggestion
Core packages may or may not decide to provide bug fix releases during the full 2 year period after release.
```

missing verb ?

Therefore, projects may occasionally want to drop support for core package dependencies earlier than recommended by this SPEC.
For instance, if a newer minimum version of a core package is needed by a project due to a critical bug fix,
which is not backported to older versions.
{{< /admonition >}}

### Core Project Endorsement

<!--
Briefly discuss what it means for a core project to endorse this SPEC.
-->

Core project endorsing this SPEC means that those projects encourage all projects across the Scientific Python ecosystem
to limit how long they support older Python versions and older core packages versions.
A core project endorsing this SPEC does **not** imply that that project will provide bug-fix releases for two full years after a release.

> As many of the core projects are volunteer projects with limited developer time, core package developers may often prioritize
> new features and enhancements over long-term package maintenance.

Author: ```suggestion
```

1. Support for Python versions be dropped **3 years** after their initial release.
2. Support for core package dependencies be dropped **2 years** after their initial release.

{{< admonition note >}}
Core packages may or may not to provide bug fix releases during the full 2 year period after release.

> So projects may decide to occassionally drop support for core pakage dependencies earlier than recommended by this SPEC.

Reviewer: ```suggestion
Therefore, projects may occassionally want to drop support for core package dependencies earlier than recommended by this SPEC.
```

For instance, if a newer minimum version of a core package is needed by a project due to a critical bug fix,
which is not backported to older versions.
{{< /admonition >}}

### Core Project Endorsement

<!--
Briefly discuss what it means for a core project to endorse this SPEC.
-->

Core project endorsing this SPEC means that those projects encourage all projects across the Scientific Python ecosystem
to limit how long they support older Python versions and older core packages versions.

> While core projects may wish to provide bug-fix releases of older versions, endorsing this SPEC does **not** imply that
> core projects will provide bug-fix releases during the full 2 year period after release.

Reviewer: ```suggestion
A core project endorsing this SPEC does **not** imply that that project will provide bug-fix releases for two full years after a release.
```

As many of the core projects are volunteer projects with limited developer time, core package developers may often prioritize

> new features and enhancements over long-term package maintenance.

Reviewer 4: I don't think we should add that note. For some projects that could be argued and here that would not really help the messaging IMHO. At most I would say something along the line of "project's developments are discussed and prioritised with and by the community".

Reviewer 2: I agree that it's better without the last sentence.
